src/anemoi/training/config/training/default.yaml

```diff
@@ -33,10 +33,34 @@ swa:
 # use ZeroRedundancyOptimizer ; saves memory for larger models
 zero_optimizer: False
 
-# dynamic rescaling of the loss gradient
-# see https://arxiv.org/pdf/2306.06079.pdf, section 4.3.2
-# don't enable this by default until it's been tested and proven beneficial
-loss_gradient_scaling: False
+# loss functions
+loss_functions:
+
+  # dynamic rescaling of the loss gradient
+  # see https://arxiv.org/pdf/2306.06079.pdf, section 4.3.2
+  # don't enable this by default until it's been tested and proven beneficial
+  loss_gradient_scaling: False
+
+
+  # loss function for the model
+  loss:
+    # loss class to initialise, can be anything subclassing torch.nn.Module
+    _target_: anemoi.training.losses.mse.WeightedMSELoss
+    # what to include in the loss class initialisation
+    include_node_weights: True
+    include_feature_weights: True
+    # other kwargs
+    ignore_nans: False
+
+  # loss function for metric calculation
+  metrics:
+    # loss class to initialise, can be anything subclassing torch.nn.Module
+    _target_: anemoi.training.losses.mse.WeightedMSELoss
+    # what to include in the loss class initialisation
+    include_node_weights: True
+    include_feature_weights: False
+    # other kwargs
+    ignore_nans: True
```

Comment:
I have a suspicion that making this configurable will end in more bugs than utility.

E.g. The metrics have to ignore Nans, because they're outside of the imputer. But ideally, a user on the config-side shouldn't have to know about this, hence why we hard-code it.

Reply:
I can understand the appeal of hard coding things that make sense, but if a user brings a loss function which exposes other configuration options, or deals with nan's another way, we need to support that configuration

Reply:
Setting sensible defaults for our losses will resolve most of the 'bugs' that might be caused by the configurablilty

Reply:
On another note here, I did consider the "ignore_nans" to be a temporary fix. We could just write two different classes here. One that handles Nans and one that doesn't, then it's handled by the initialisation instead of bloating the function signature.

As a user you can then explicitly choose between the NanWeightedRMSE or the WeightedRMSE, which would now be possible with this implementation.

Reply:
> On another note here, I did consider the "ignore_nans" to be a temporary fix. We could just write two different classes here. One that handles Nans and one that doesn't, then it's handled by the initialisation instead of bloating the function signature.
> 
> As a user you can then explicitly choose between the NanWeightedRMSE or the WeightedRMSE, which would now be possible with this implementation.

Yep on the ignore_nans being temporary, I think the initial motivation for having nan_mean and nan_sum in the loss function is also being solved for by PR #72  (pass masks to the loss containing info through where we expect to see nans). 

Unless there was another usecase for the nan_mean/nan_sum, I think they won't be needed after/if PR #72 passes?

Reply:
> I think they won't be needed after/if PR https://github.com/ecmwf/anemoi-training/pull/72 passes?

Okay, thanks @Rilwan-Adewoyin. Could it be PR #72 role to remove the `nan_mean`?